src/components/field-label/field-label.tsx

```diff
@@ -14,11 +14,11 @@ type FieldLabelProps = {
   hint: React.ReactNode;
   description: React.ReactNode;
   onInfoButtonClick?: () => void;
-  hintIcon: React.ReactNode;
+  hintIcon: React.ReactElement<any>;
```

Comment:
This is due to the fact that we `createElement` it. Hence it needs to be an element. In a way it's a more strict typing than with prop-types.